exe/dirtree

```diff
@@ -84,6 +93,18 @@ end
 
 if options.key?(:output)
   File.write(options[:output], result)
+elsif options.key?(:screenshot)
```

Comment:
here there is a change in the general idea or how `dirtree` works, the old behavior will write the result to a file if specified, otherwise it'll be written to the standard output.

adding the screenshot feature means this old logic isn't suitable anymore, as you have 2 options writing to a file HTML output, writing to a file screenshot, and both are not mutually exclusive, so a user can write both.

that mean we can't have an `if` `else` here, and one of the two cases should be implemented :  
- the option to write to the standard output isn't suitable anymore and should be removed
- the option to write to the standard output should be executed if both `output` and `screenshot` options are not provided.

I advise going for the second option, the the logic should be 

```ruby
File.write(options[:output], result) if options.key?(:output)

if options.key?(:screenshot)
<writing screenshot data here>
end

unless option.key?(:output) || option.key?(:screenshot)
   puts result
end
```

Reply:
I wanted to remove any redundant I/O calls.
I think it won't harm if I checked whether the user used the output flag or not.

```ruby
File.write(options[:output], result) if options.key?(:output)

if options.key?(:screenshot)
  unless options.key(:output)?
    tmp = Tempfile.new(default_file_name)
    tmp.write(result)
    tmp.close
  end
  file = options[:output] || tmp.path
  %w(chromium chrome chrome-stable).each |chrome| do
    system(chrome, '--disable-gpu', '--headless', '--screenshot', file)
  end
end

unless option.key?(:output) || option.key?(:screenshot)
   puts result
end
```

Reply:
I understand, let's just make sure we have the correct logic, Iagree on not writing the file twice.
but review your logic in that snippet, specially the `system` line as it will continue to execute even after finding the correct chrome executable.

Reply:
```ruby    
screenshot = system(chrome, '--disable-gpu', '--headless', '--screenshot', file)
break if screenshot
```